src/DataGrid.tsx

```diff
@@ -1196,7 +1196,10 @@ function DataGrid<R, SR, K extends Key>(
               selectedPosition.rowIdx
             ),
             [rowSelected]: isGroupRowFocused,
-            [rowSelectedWithFrozenCell]: isGroupRowFocused && lastFrozenColumnIndex !== -1
+            [rowSelectedWithFrozenCell]:
+              isGroupRowFocused &&
+              lastFrozenColumnIndex !== -1 &&
+              selectedPosition.rowIdx !== maxRowIdx
```

Comment:
This change breaks the selected row style when columns are scrolled. `:before` is used to set the left border 
![image](https://github.com/user-attachments/assets/7dc70c9a-1ff3-4a42-a011-392ede326529)